pkg/server/server.go

```diff
@@ -105,9 +112,50 @@ func (s *server) RunUntil(stopCh <-chan struct{}) error {
 		return nil
 	}
 
+	g := run.Group{}
 	// Start serving API and scrape loop
-	go s.runScrape(ctx)
-	return s.GenericAPIServer.PrepareRun().Run(stopCh)
+	{
```

Comment:
Can you create separate functions instead of using comments and blocks?

Reply:
Sounds good, done :+1: